module-4/backend/week_two.md

```diff
@@ -5,18 +5,55 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's one difference between ES5 and ES6?
+
+- ES6 allows you to organize with class and a constructor
+
 2. What's the difference between asynchronous and synchronous JavaScript? 
+
+- Synchronous runs in order but asynchronous runs concurrently
+
 3. What are the four pillars of Object Oriented programming?
+
+- Encapsulation, Polymorphism, Inheritance, Abstraction
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+
+- export/import, class/constructor
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+
+- SRP, DRY, consistency (especially in using es5/es6)
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+
+- When a function calls another function as an argument. Needing to pass the return or object / this that is being acted upon to the next function
+
 7. What's the scope of variables in Javascript?
+
+- Can access parent variables but not children's. Can edit parent varibles from within child scope. Can't access another function's variables.
+
 8. What's the difference between `==` and `===` in JavaScript?
+
+- '===' looks for equivalency, '==' looks for same type of data
+
 9. Why do front end frameworks exist?
 
+- To organize code in MVC or MVVM way for example, as a builder tool to deploy, and/or to have a certain design around how the code interacts
+
 #### Review  
 
 10. Why do people say "HTTP is stateless"?
+
+- It doesn't hold state, making a request to the server and the response doesn't hold state, we can use cookies for example to allow client to hold a state
+
 11. Describe a RESTful API.
+
+- Naming conventions around CRUD functionality of data. Allows you to create, edit, update, delete, read all, read one record's data
```

Comment:
it also follows conventions on which HTTP verb is used when accomplishing a CRUD task along with constructing the url path accordingly